truffle/com.oracle.truffle.api.test/src/com/oracle/truffle/api/vm/EngineTest.java

```diff
@@ -117,4 +119,19 @@ public void wrappedAsArray() throws Exception {
         assertEquals(2, arr[1].intValue());
         assertEquals(3, arr[2].intValue());
     }
+
+    @Test
+    public void initializePolyglotEngineWithArguments() throws IOException {
+        PolyglotEngine vm = createBuilder().setArguments(ExportImportLanguage1.class, new String[]{"1", "2"}).build();
```

Comment:
Ok, used the MIME type. What's the reason for this? Potential class loading issues or something? Intuitively, I would assume that this kind of interaction still happens in tightly bound places, like a main method or something similar. You envision a more dynamic and modular use case?